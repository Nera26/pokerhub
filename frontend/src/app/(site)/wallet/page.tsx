import dynamic from 'next/dynamic';

const WalletPage = dynamic(() => import('@/features/site/wallet'), {
  loading: () => <div>Loading...</div>,
});

export default WalletPage;
