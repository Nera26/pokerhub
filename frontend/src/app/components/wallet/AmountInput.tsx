'use client';

import Input, { type InputProps } from '../ui/Input';

export interface AmountInputProps extends Omit<InputProps, 'type'> {}

export default function AmountInput(props: AmountInputProps) {
  return <Input type="number" placeholder="0.00" {...props} />;
}

