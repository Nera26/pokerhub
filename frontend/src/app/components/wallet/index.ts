export { default as AmountInput } from './AmountInput';
export { default as BankTransferForm } from './BankTransferForm';
