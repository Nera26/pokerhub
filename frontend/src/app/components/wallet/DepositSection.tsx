'use client';

import { useState, useEffect, useRef } from 'react';
import { useForm } from 'react-hook-form';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import Tooltip from '../ui/Tooltip';
import Button from '../ui/Button';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';
import { useIban } from '@/hooks/wallet';

export interface DepositSectionProps {
  /** Called when user clicks close icon */
  onClose: () => void;
  /** Called when user confirms they have sent the deposit */
  onConfirm: () => void;
}

export default function DepositSection({
  onClose,
  onConfirm,
}: DepositSectionProps) {
  const depositSchema = z.object({
    confirm: z
      .boolean()
      .refine((v) => v, { message: 'Please confirm the deposit' }),
  });
  type DepositForm = z.infer<typeof depositSchema>;

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<DepositForm>({
    resolver: zodResolver(depositSchema),
    defaultValues: { confirm: false },
  });

  const [countdown, setCountdown] = useState(10);
  const timerRef = useRef<number | null>(null);
  const { data, isLoading, error } = useIban();

  useEffect(() => {
    // start countdown
    timerRef.current = window.setInterval(() => {
      setCountdown((c) => {
        if (c <= 1) {
          if (timerRef.current !== null) {
            clearInterval(timerRef.current);
          }
          return 0;
        }
        return c - 1;
      });
    }, 1000);
    return () => {
      if (timerRef.current !== null) {
        clearInterval(timerRef.current);
      }
    };
  }, []);

  const handleCopy = async () => {
    if (!data?.iban) {
      alert('Failed to copy account number');
      return;
    }
    try {
      await navigator.clipboard.writeText(data.iban);
      alert('Account number copied to clipboard');
    } catch {
      alert('Failed to copy account number');
    }
  };

  const onSubmit = handleSubmit(() => onConfirm());

  return (
    <form
      onSubmit={onSubmit}
      className="max-h-[90vh] overflow-y-auto"
      noValidate
    >
      {/* Header */}
      <div className="flex justify-between items-center mb-6">
        <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
          Deposit via Bank Transfer
        </h2>
        <button
          onClick={onClose}
          className="text-text-secondary hover:text-accent-yellow text-2xl"
          aria-label="Close deposit modal"
        >
          <FontAwesomeIcon icon={faTimes} />
        </button>
      </div>

      {/* Instructions */}
      <div className="bg-primary-bg rounded-xl p-4 mb-6 space-y-1 text-text-secondary text-sm">
        <p>
          <span className="font-semibold">Bank Name:</span> Trade &amp; Trust
          Bank
        </p>
        <p>
          <span className="font-semibold">Account Name:</span> PokerHub LLC
        </p>
        <p>
          <span className="font-semibold">Bank Address:</span> Bayanzurkh
          District, Ulaanbaatar
        </p>
        <p className="flex items-center">
          <span className="font-semibold">Account Number:</span>
          {isLoading ? (
            <span className="ml-2">Loading IBAN...</span>
          ) : error ? (
            <span className="ml-2 text-danger-red">{error.message}</span>
          ) : (
            <Tooltip text="Click to copy">
              <span
                onClick={handleCopy}
                className="ml-2 font-medium cursor-pointer"
              >
                {data?.masked}
              </span>
            </Tooltip>
          )}
        </p>
        <p className="italic text-xs mt-2">
          After transfer, your balance will be updated within 1–3 minutes.
        </p>
      </div>

      <div className="mb-6">
        <label className="flex items-center gap-2">
          <input type="checkbox" className="rounded" {...register('confirm')} />
          <span className="text-sm text-text-secondary">
            I confirm that I have sent the deposit
          </span>
        </label>
        {errors.confirm && (
          <p className="text-xs text-danger-red mt-1">
            {errors.confirm.message}
          </p>
        )}
      </div>

      {/* Confirm Button */}
      <Button
        type="submit"
        variant="primary"
        className="w-full uppercase py-3 text-sm sm:text-base disabled:opacity-50 disabled:cursor-not-allowed"
        disabled={countdown > 0}
      >
        {countdown > 0 ? `Waiting ${countdown}s…` : "I've Sent the Deposit"}
      </Button>
    </form>
  );
}
