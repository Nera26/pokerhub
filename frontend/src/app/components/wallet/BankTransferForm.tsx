'use client';

import { ReactNode } from 'react';
import { useForm } from 'react-hook-form';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import Button from '../ui/Button';
import AmountInput from './AmountInput';

export interface BankTransferFormProps {
  currency: string;
  submitLabel: string;
  amountInputId: string;
  onSubmit: (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => void | Promise<void>;
  maxAmount?: number;
  children?: ReactNode;
}

export default function BankTransferForm({
  currency,
  submitLabel,
  amountInputId,
  onSubmit,
  maxAmount,
  children,
}: BankTransferFormProps) {
  const schema = z.object({
    amount: z
      .string()
      .refine(
        (val) => {
          const num = Number(val);
          return val !== '' && !Number.isNaN(num) && num > 0;
        },
        { message: 'Enter a valid amount' },
      )
      .refine(
        (val) =>
          maxAmount === undefined ? true : Number(val) <= maxAmount,
        { message: 'Insufficient funds' },
      ),
  });
  type Form = z.infer<typeof schema>;

  const {
    register,
    handleSubmit,
    formState: { errors, isValid },
  } = useForm<Form>({
    resolver: zodResolver(schema),
    defaultValues: { amount: '' },
    mode: 'onChange',
  });

  const getDeviceId = () => {
    let id = localStorage.getItem('deviceId');
    if (!id) {
      id = crypto.randomUUID();
      localStorage.setItem('deviceId', id);
    }
    return id;
  };

  const submit = handleSubmit((data) =>
    onSubmit({
      amount: Number(data.amount),
      deviceId: getDeviceId(),
      currency,
    }),
  );

  return (
    <form onSubmit={submit} className="space-y-4" noValidate>
      <AmountInput
        id={amountInputId}
        label={`Enter Amount (${currency})`}
        error={errors.amount?.message}
        {...register('amount')}
      />
      {children}
      <Button
        type="submit"
        variant="primary"
        className="w-full uppercase py-3"
        disabled={!isValid}
      >
        {submitLabel}
      </Button>
    </form>
  );
}

