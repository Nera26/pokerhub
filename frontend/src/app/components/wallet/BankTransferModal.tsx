'use client';

import { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';
import Tooltip from '../ui/Tooltip';
import Button from '../ui/Button';
import BankTransferForm from './BankTransferForm';
import { BankTransferDepositResponse } from '@shared/wallet.schema';

interface BaseProps {
  mode: 'deposit' | 'withdraw';
  onClose: () => void;
  currency: string;
  onSubmit: (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => void | Promise<BankTransferDepositResponse | void>;
}

interface DepositProps extends BaseProps {
  mode: 'deposit';
}

interface WithdrawProps extends BaseProps {
  mode: 'withdraw';
  availableBalance: number;
  bankDetails: {
    bankName: string;
    accountName: string;
    bankAddress: string;
    maskedAccountNumber: string;
  };
}

export type BankTransferModalProps = DepositProps | WithdrawProps;

export default function BankTransferModal(props: BankTransferModalProps) {
  const { mode, onClose, currency, onSubmit } = props;
  const [details, setDetails] = useState<BankTransferDepositResponse | null>(
    null,
  );

  const handleSubmit = async (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => {
    const res = await onSubmit(payload);
    if (mode === 'deposit') {
      setDetails(res as BankTransferDepositResponse);
    }
  };

  const handleCopy = async () => {
    if (!details) return;
    try {
      await navigator.clipboard.writeText(details.bank.accountNumber);
    } catch {
      // ignore
    }
  };

  if (mode === 'deposit' && details) {
    return (
      <div className="space-y-4">
        <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
          Bank Transfer Instructions
        </h2>
        <div className="bg-primary-bg rounded-xl p-4 space-y-1 text-text-secondary text-sm">
          <p>
            <span className="font-semibold">Bank Name:</span>{' '}
            {details.bank.bankName}
          </p>
          <p className="flex items-center">
            <span className="font-semibold">Account Number:</span>
            <Tooltip text="Click to copy">
              <span
                onClick={handleCopy}
                className="ml-2 font-medium cursor-pointer"
              >
                {details.bank.accountNumber}
              </span>
            </Tooltip>
          </p>
          <p>
            <span className="font-semibold">Routing Code:</span>{' '}
            {details.bank.routingCode}
          </p>
          <p>
            <span className="font-semibold">Reference:</span>{' '}
            {details.reference}
          </p>
        </div>
        <Button
          type="button"
          variant="primary"
          className="w-full uppercase py-3"
          onClick={onClose}
        >
          Close
        </Button>
      </div>
    );
  }

  return (
    <div
      className={
        mode === 'withdraw' ? 'max-h-[90vh] overflow-y-auto' : 'space-y-4'
      }
    >
      {mode === 'withdraw' ? (
        <div className="flex justify-between items-center mb-6">
          <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
            Withdraw Funds
          </h2>
          <button
            onClick={onClose}
            className="text-text-secondary hover:text-accent-yellow text-2xl"
            aria-label="Close withdraw modal"
          >
            <FontAwesomeIcon icon={faTimes} />
          </button>
        </div>
      ) : (
        <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
          Deposit via Bank Transfer
        </h2>
      )}

      <div className={mode === 'withdraw' ? 'mb-4' : ''}>
        <BankTransferForm
          currency={currency}
          submitLabel={mode === 'deposit' ? 'Get Instructions' : 'Withdraw'}
          amountInputId={
            mode === 'deposit' ? 'deposit-amount' : 'withdraw-amount'
          }
          onSubmit={handleSubmit}
          maxAmount={mode === 'withdraw' ? props.availableBalance : undefined}
        >
          {mode === 'withdraw' && (
            <p className="text-xs text-text-secondary mt-1">
              Available: {props.availableBalance.toFixed(2)} {currency}
            </p>
          )}
        </BankTransferForm>
      </div>

      {mode === 'withdraw' && props.bankDetails && (
        <div className="mb-6 bg-primary-bg rounded-xl p-4">
          <p className="text-text-secondary mb-1">
            <span className="font-semibold">Bank Name:</span>{' '}
            {props.bankDetails.bankName}
          </p>
          <p className="text-text-secondary mb-1">
            <span className="font-semibold">Account Name:</span>{' '}
            {props.bankDetails.accountName}
          </p>
          <p className="text-text-secondary mb-1">
            <span className="font-semibold">Bank Address:</span>{' '}
            {props.bankDetails.bankAddress}
          </p>
          <p className="text-text-secondary">
            <span className="font-semibold">Account Number:</span>{' '}
            {props.bankDetails.maskedAccountNumber}
          </p>
        </div>
      )}
    </div>
  );
}
