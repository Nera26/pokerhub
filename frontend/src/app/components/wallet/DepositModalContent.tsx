'use client';

import { useState } from 'react';
import Button from '../ui/Button';
import { BankTransferDepositResponse } from '@shared/wallet.schema';
import Tooltip from '../ui/Tooltip';
import BankTransferForm from './BankTransferForm';

export interface DepositModalContentProps {
  onClose: () => void;
  onInitiate: (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => Promise<BankTransferDepositResponse>;
  currency: string;
}

export default function DepositModalContent({
  onClose,
  onInitiate,
  currency,
}: DepositModalContentProps) {
  const [details, setDetails] = useState<BankTransferDepositResponse | null>(null);

  const handleSubmit = async (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => {
    const res = await onInitiate(payload);
    setDetails(res);
  };

  const handleCopy = async () => {
    if (!details) return;
    try {
      await navigator.clipboard.writeText(details.bank.accountNumber);
    } catch {
      // ignore
    }
  };

  if (details) {
    return (
      <div className="space-y-4">
        <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
          Bank Transfer Instructions
        </h2>
        <div className="bg-primary-bg rounded-xl p-4 space-y-1 text-text-secondary text-sm">
          <p>
            <span className="font-semibold">Bank Name:</span> {details.bank.bankName}
          </p>
          <p className="flex items-center">
            <span className="font-semibold">Account Number:</span>
            <Tooltip text="Click to copy">
              <span
                onClick={handleCopy}
                className="ml-2 font-medium cursor-pointer"
              >
                {details.bank.accountNumber}
              </span>
            </Tooltip>
          </p>
          <p>
            <span className="font-semibold">Routing Code:</span> {details.bank.routingCode}
          </p>
          <p>
            <span className="font-semibold">Reference:</span> {details.reference}
          </p>
        </div>
        <Button
          type="button"
          variant="primary"
          className="w-full uppercase py-3"
          onClick={onClose}
        >
          Close
        </Button>
      </div>
    );
  }

  return (
    <div className="space-y-4">
      <h2 className="text-xl sm:text-2xl font-bold text-text-primary">
        Deposit via Bank Transfer
      </h2>
      <BankTransferForm
        currency={currency}
        submitLabel="Get Instructions"
        amountInputId="deposit-amount"
        onSubmit={handleSubmit}
      />
    </div>
  );
}

