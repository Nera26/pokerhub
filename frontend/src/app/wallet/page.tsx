'use client';

import { useEffect, useState } from 'react';
import BankTransferModal from '../components/wallet/BankTransferModal';
import Button from '../components/ui/Button';
import ToastNotification from '../components/ui/ToastNotification';
import InlineError from '../components/ui/InlineError';
import useToasts from '@/hooks/useToasts';
import { useApiError } from '@/hooks/useApiError';
import { useBankTransfer, useWalletStatus, useWithdraw } from '@/hooks/wallet';

export default function WalletPage() {
  const [mode, setMode] = useState<'deposit' | 'withdraw' | null>(null);
  const { data: status } = useWalletStatus();
  const bankTransfer = useBankTransfer();
  const withdraw = useWithdraw();
  const { toasts, pushToast } = useToasts();

  const activeMutation = mode === 'withdraw' ? withdraw : bankTransfer;
  const errorMessage = useApiError(activeMutation.error);

  useEffect(() => {
    if (bankTransfer.isSuccess)
      pushToast('Bank transfer initiated', { variant: 'success' });
  }, [bankTransfer.isSuccess, pushToast]);

  useEffect(() => {
    if (withdraw.isSuccess)
      pushToast('Withdrawal requested', { variant: 'success' });
  }, [withdraw.isSuccess, pushToast]);

  const handleSubmit = (payload: {
    amount: number;
    deviceId: string;
    currency: string;
  }) => {
    return mode === 'withdraw'
      ? withdraw.mutateAsync(payload)
      : bankTransfer.mutateAsync(payload);
  };

  return (
    <>
      <main className="p-4 space-y-4">
        <div className="flex gap-4">
          <Button onClick={() => setMode('deposit')}>Deposit</Button>
          <Button onClick={() => setMode('withdraw')}>Withdraw</Button>
        </div>

        {mode && status && (
          <div
            aria-busy={activeMutation.isPending}
            className={
              activeMutation.isPending ? 'pointer-events-none opacity-50' : ''
            }
          >
            <BankTransferModal
              mode={mode}
              onClose={() => setMode(null)}
              currency={status.currency}
              onSubmit={handleSubmit}
              {...(mode === 'withdraw'
                ? {
                    availableBalance: status.realBalance,
                    bankDetails: {
                      bankName: 'Your Bank',
                      accountName: 'John Doe',
                      bankAddress: '123 Street',
                      maskedAccountNumber: '****1234',
                    },
                  }
                : {})}
            />
            {errorMessage && <InlineError message={errorMessage} />}
          </div>
        )}
      </main>

      {toasts.map((t) => (
        <ToastNotification
          key={t.id}
          message={t.message}
          type={t.variant === 'error' ? 'error' : 'success'}
          isOpen
          duration={t.duration}
          onClose={() => {}}
        />
      ))}
    </>
  );
}
