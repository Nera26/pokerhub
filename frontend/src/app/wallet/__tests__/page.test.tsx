import { render, screen } from '@testing-library/react';
import WalletPage from '../page';
import { useWalletStatus } from '@/hooks/wallet';

jest.mock('@/hooks/wallet', () => ({
  useWalletStatus: jest.fn(),
}));

describe('WalletPage', () => {
  beforeEach(() => {
    (useWalletStatus as jest.Mock).mockReset();
  });

  it('shows loading state', () => {
    (useWalletStatus as jest.Mock).mockReturnValue({
      data: undefined,
      isLoading: true,
      error: null,
    });
    render(<WalletPage />);
    expect(screen.getByText(/loading wallet/i)).toBeInTheDocument();
  });

  it('shows error state', () => {
    (useWalletStatus as jest.Mock).mockReturnValue({
      data: undefined,
      isLoading: false,
      error: new Error('fail'),
    });
    render(<WalletPage />);
    expect(screen.getByRole('alert')).toHaveTextContent(
      /failed to load wallet/i,
    );
  });

  it('shows empty state', () => {
    (useWalletStatus as jest.Mock).mockReturnValue({
      data: undefined,
      isLoading: false,
      error: null,
    });
    render(<WalletPage />);
    expect(screen.getByText(/no wallet data/i)).toBeInTheDocument();
  });

  it('matches snapshot when KYC verified', () => {
    (useWalletStatus as jest.Mock).mockReturnValue({
      data: {
        realBalance: 100,
        creditBalance: 50,
        kycVerified: true,
        currency: 'USD',
      },
      isLoading: false,
      error: null,
    });
    const { container } = render(<WalletPage />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when KYC pending', () => {
    (useWalletStatus as jest.Mock).mockReturnValue({
      data: {
        realBalance: 100,
        creditBalance: 50,
        kycVerified: false,
        currency: 'USD',
      },
      isLoading: false,
      error: null,
    });
    const { container } = render(<WalletPage />);
    expect(container).toMatchSnapshot();
  });
});
