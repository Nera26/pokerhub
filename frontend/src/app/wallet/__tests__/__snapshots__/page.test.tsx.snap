// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPage matches snapshot when KYC pending 1`] = `
<div>
  <section
    class="bg-card-bg rounded-2xl p-8 md:p-12 mb-6 md:mb-8"
  >
    <div
      class="flex flex-col md:flex-row md:items-center justify-between"
    >
      <div>
        <p
          class="text-text-secondary text-sm sm:text-base"
        >
          Total Balance
        </p>
        <p
          class="text-3xl sm:text-4xl font-bold text-accent-yellow mb-1"
        >
          $150.00
        </p>
        <p
          class="text-xs text-text-secondary mb-2"
        >
          KYC Status: 
          Pending
        </p>
        <p
          class="text-sm sm:text-base text-text-secondary"
        >
          <span
            class="font-medium"
          >
            • Real: 
            $100.00
          </span>
           | 
          <span
            aria-describedby="«r1»"
            class="relative inline-block cursor-pointer"
            tabindex="0"
          >
            <span
              class="font-medium"
            >
              • Credit: 
              $50.00
            </span>
            <span
              class="absolute bottom-full left-1/2 mb-2 w-max max-w-xs whitespace-nowrap rounded bg-card-bg px-2 py-1 text-xs text-text-primary z-50 transition-opacity duration-200 opacity-0 invisible"
              id="«r1»"
              role="tooltip"
            >
              Credits can only be used for games and cannot be withdrawn.
            </span>
          </span>
        </p>
      </div>
      <div
        class="mt-6 md:mt-0 flex space-x-4"
      >
        <button
          class="justify-center gap-2 rounded-xl font-semibold transition-colors duration-200 disabled:opacity-50 disabled:cursor-not-allowed select-none focus:outline-none focus-glow-yellow px-6 py-3 bg-transparent border border-accent-yellow text-accent-yellow hover:bg-accent-yellow hover:text-primary-bg text-sm sm:text-base uppercase flex items-center"
          data-variant="outline"
          type="button"
        >
          <span
            class="inline-block"
          >
            Verify
          </span>
        </button>
        <button
          class="justify-center gap-2 rounded-xl font-semibold transition-colors duration-200 disabled:opacity-50 disabled:cursor-not-allowed select-none focus:outline-none focus-glow-yellow px-6 py-3 bg-accent-green text-text-primary hover-glow-green text-sm sm:text-base uppercase flex items-center"
          data-variant="primary"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down mr-2"
            data-icon="arrow-down"
            data-prefix="fas"
            role="img"
            viewBox="0 0 384 512"
          >
            <path
              d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7-105.4-105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
          <span
            class="inline-block"
          >
             Deposit
          </span>
        </button>
        <button
          class="justify-center gap-2 rounded-xl font-semibold transition-colors duration-200 disabled:opacity-50 disabled:cursor-not-allowed select-none focus:outline-none focus-glow-yellow px-6 py-3 bg-transparent border border-accent-yellow text-accent-yellow hover:bg-accent-yellow hover:text-primary-bg text-sm sm:text-base uppercase flex items-center"
          data-variant="outline"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up mr-2"
            data-icon="arrow-up"
            data-prefix="fas"
            role="img"
            viewBox="0 0 384 512"
          >
            <path
              d="M214.6 17.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 117.3 160 488c0 17.7 14.3 32 32 32s32-14.3 32-32l0-370.7 105.4 105.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
          <span
            class="inline-block"
          >
             Withdraw
          </span>
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`WalletPage matches snapshot when KYC verified 1`] = `
<div>
  <section
    class="bg-card-bg rounded-2xl p-8 md:p-12 mb-6 md:mb-8"
  >
    <div
      class="flex flex-col md:flex-row md:items-center justify-between"
    >
      <div>
        <p
          class="text-text-secondary text-sm sm:text-base"
        >
          Total Balance
        </p>
        <p
          class="text-3xl sm:text-4xl font-bold text-accent-yellow mb-1"
        >
          $150.00
        </p>
        <p
          class="text-xs text-text-secondary mb-2"
        >
          KYC Status: 
          Verified
        </p>
        <p
          class="text-sm sm:text-base text-text-secondary"
        >
          <span
            class="font-medium"
          >
            • Real: 
            $100.00
          </span>
           | 
          <span
            aria-describedby="«r0»"
            class="relative inline-block cursor-pointer"
            tabindex="0"
          >
            <span
              class="font-medium"
            >
              • Credit: 
              $50.00
            </span>
            <span
              class="absolute bottom-full left-1/2 mb-2 w-max max-w-xs whitespace-nowrap rounded bg-card-bg px-2 py-1 text-xs text-text-primary z-50 transition-opacity duration-200 opacity-0 invisible"
              id="«r0»"
              role="tooltip"
            >
              Credits can only be used for games and cannot be withdrawn.
            </span>
          </span>
        </p>
      </div>
      <div
        class="mt-6 md:mt-0 flex space-x-4"
      >
        <button
          class="justify-center gap-2 rounded-xl font-semibold transition-colors duration-200 disabled:opacity-50 disabled:cursor-not-allowed select-none focus:outline-none focus-glow-yellow px-6 py-3 bg-accent-green text-text-primary hover-glow-green text-sm sm:text-base uppercase flex items-center"
          data-variant="primary"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down mr-2"
            data-icon="arrow-down"
            data-prefix="fas"
            role="img"
            viewBox="0 0 384 512"
          >
            <path
              d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7-105.4-105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
          <span
            class="inline-block"
          >
             Deposit
          </span>
        </button>
        <button
          class="justify-center gap-2 rounded-xl font-semibold transition-colors duration-200 disabled:opacity-50 disabled:cursor-not-allowed select-none focus:outline-none focus-glow-yellow px-6 py-3 bg-transparent border border-accent-yellow text-accent-yellow hover:bg-accent-yellow hover:text-primary-bg text-sm sm:text-base uppercase flex items-center"
          data-variant="outline"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up mr-2"
            data-icon="arrow-up"
            data-prefix="fas"
            role="img"
            viewBox="0 0 384 512"
          >
            <path
              d="M214.6 17.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 117.3 160 488c0 17.7 14.3 32 32 32s32-14.3 32-32l0-370.7 105.4 105.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
          <span
            class="inline-block"
          >
             Withdraw
          </span>
        </button>
      </div>
    </div>
  </section>
</div>
`;
